src/yadism/coefficient_functions/fonll/g1_nc.py

```diff
@@ -57,38 +57,19 @@ def NNLO(self):
         return RSL(raw.c2g_NNLL_reg, args=[self.L])
 
 
-class PdfMatchingLLNonSinglet(pc.PartonicChannelAsy):
-    def NNLO(self):
-        # NOTE: the LL contribution is the same as the unpolarized one.
-        return RSL(
-            pc.pdf_matching_LL_reg,
-            pc.pdf_matching_LL_sing,
-            pc.pdf_matching_LL_loc,
-            args=[self.L],
-        )
-
+# NOTE: the qq NS matching is the same as the unpolarized one due to
+# Ward identites.
+class PdfMatchingLLNonSinglet(pc.PdfMatchingLLNonSinglet):
+   pass
 
-class PdfMatchingNLLNonSinglet(pc.PartonicChannelAsy):
-    def NNLO(self):
-        # NOTE: the NLL contribution is the same as the unpolarized one.
-        return RSL(
-            sing=pc.pdf_matching_NLL_sing,
-            loc=pc.pdf_matching_NLL_loc,
-            args=[self.L],
-        )
 
+class PdfMatchingNLLNonSinglet(pc.PdfMatchingNLLNonSinglet):
+    pass
 
-class PdfMatchingNNLLNonSinglet(pc.PartonicChannelAsy):
-    def NNLO(self):
-        # NOTE: the NNLL contribution is differs from the unpolarized one.
-        return RSL(
-            raw.pdf_matching_NNLL_reg,
-            raw.pdf_matching_NNLL_sing,
-            raw.pdf_matching_NNLL_loc,
-        )
-
+class PdfMatchingNNLLNonSinglet(pc.PdfMatchingNNLLNonSinglet):
+    pass
 
-class PdfMatchingNNNLLNonSinglet(pc.PartonicChannelAsy):
+class PdfMatchingNNNLLNonSinglet(pc.PdfMatchingNNNLLNonSinglet):
     pass
```

Comment:
this might be a left-over from when we had the mismatch of the log order (do you remember @giacomomagni ?) - there should be no N3LL stuff (neither here nor for unpolarized); and indeed they are empty. These would become relevant with FONLL-D/E/F. Please delete both

Reply:
wait that was added to make the N3LO working, because of this: https://github.com/NNPDF/yadism/blob/49446b1980b9657131f67d91a4043d993a3b223b/src/yadism/coefficient_functions/fonll/kernels.py#L83-L90

Reply:
Okay, okay, you're right we need it! Exactly as I said "relevant with FONLL-D/E/F" sorry